### PR TITLE
[Feature] 8720 toolbar at bottom of details view

### PR DESCRIPTION
**Pull request by hschink**
[`* `#8720` Toolbar at bottom detail view`](https://www.openproject.org/work_packages/8720)

:exclamation: https://github.com/finnlabs/openproject-costs/pull/104

### CODE REVIEW

File: app/assets/javascripts/angular/services/work-package-service.js [hunk 1]
```diff
@@ -179,13 +182,36 @@ angular.module('openproject.services')
...
+      }
+
       var params = {
-        'ids[]': workPackages.map(function(wp) {
-          return wp.id;
-        })
+        'ids[]': ids
       };
-      return $http['delete'](PathHelper.workPackagesBulkDeletePath(), { params: params });
+      var promis = $http['delete'](PathHelper.workPackagesBulkDeletePath(), { params: params });
```
Comment (myabc) [inline app/assets/javascripts/angular/services/work-package-service.js]: @hschink is `promis` a typo or intentional?

File: lib/api/v3/work_packages/work_package_representer.rb [hunk 2]
```diff
@@ -64,6 +64,35 @@ def initialize(model, options = {}, *expand)
...
+            method: :delete,
+            title: "Delete #{represented.subject}"
+          } if current_user_allowed_to(:delete_work_packages, represented.model)
+        end
+
+        link :log_time do
+          {
+            href: new_work_package_time_entry_path(represented.model),
+            title: "Log time on #{represented.subject}"
+          } if current_user_allowed_to(:log_time, represented.model)
```
Comment (myabc) [inline lib/api/v3/work_packages/work_package_representer.rb:80]: @hschink where links are to "classic" HTML pages and not to other API resources, I think we should specify the MIME type, i.e. `type: 'text/html'`. See http://tools.ietf.org/html/draft-kelly-json-hal-06#section-5.3

File: app/assets/javascripts/angular/services/work-package-service.js (hunk 1 above)
Comment (hschink) [inline app/assets/javascripts/angular/services/work-package-service.js]: @myabc it is a typo. Thanks for the hint! Fixed in https://github.com/opf/openproject/commit/6529defe85d20d35e60bb4382ba23a81b8eb8f9b.

File: lib/api/v3/work_packages/work_package_representer.rb (hunk 2 above)
Comment (hschink) [inline lib/api/v3/work_packages/work_package_representer.rb:80]: @myabc thanks for the reference. Fixed in https://github.com/opf/openproject/commit/787f62d177235304f56f56dd7fdd6f0592c35181.

**Comment by myabc** [comment]
Works for me. Just a small design criticism: As the bottom toolbar is left-aligned, I think it would make more sense to open the menu to the right. WDYT?

seeded_project_-_openproject

**Comment by myabc** [comment]
The _Edit_ and _More_ buttons also have slightly different heights (25px vs. 23px). We should normalise these / @zzmedia

**Comment by hschink** [comment]
@myabc I'm fine with the context menu position. We may change the position when the requirements for positioning are clear.

**Comment by myabc** [comment]
@hschink I went ahead and merged. We'll try to address some of these relatively minor issues next week, when @zzmedia is available.

**Pull request merged**